Backbone/DaoMsgDef.py

```diff
@@ -0,0 +1,503 @@
+    # convert the bytes back into useable variables / structures
+    player_name = name_bytes.decode()
+    player_color = color_bytes.decode()
+    return player_name, player_color
+
+
+# This function extracts payload data for the challenge_response msg
+def extract_challenge_response_msg(msg : StandardDaoMsg):
+    """
+    This extracts data from a challenge_response Dao message.
+        - player_name:      The name of the player responding to a challenge
+        - accepted_status:  True if the player is accepting the challenge, false if rejecting
+    """
+    # get the bytes for each variable in the payload
+    name_bytes, accepted_btyes = splitBytesOnDelimiter(msg.payload)
+    # convert the bytes back into useable variables / structures
+    player_name = name_bytes.decode()
+    accepted_status = bool.from_bytes(accepted_btyes)
+    return player_name, accepted_status
+
+
+# This function extracts payload data for the draw_response msg
+def extract_draw_response_msg(msg : StandardDaoMsg):
+    """
+    This extracts data from a draw_resonse Dao message.
+        - accepted_status:  True if the player is accepting the draw, false if rejecting
+    """
+    # convert the payload back into useable variables / structures
+    accepted_status = bool.from_bytes(msg.payload)
+    return accepted_status
+
+
+# This function extracts payload data for the legal_move msg
+def extract_legal_move_msg(msg : StandardDaoMsg):
+    """
+    This extracts data from a legal_move Dao message.
+    - dao_piece:    An identifier to know what piece is moving
+    - new_position: The board position to move the piece to
+    """
+    # get the bytes for each variable in the payload
+    piece_bytes, posit_bytes = splitBytesOnDelimiter(msg.payload)
+    # convert the bytes back into useable variables / structures
+    # TODO: implement the reverse byte conversion once no longer using tempData structures
+    dao_piece = tempData()
+    new_position = tempData()
+    return dao_piece, new_position
+
+
+# This function extracts payload data for the end_of_game msg
+def extract_end_of_game_mgs(msg : StandardDaoMsg):
+    """
+    This extracts data from an end_of_game Dao message.
+        - end_status:  Describes the reason the game is ending
+        - win_player:  Contains the name of the winning player if there is one
+        - win_method:  Describes how the player won if there is a winning player
+    """
+    # get the bytes for each variable in the payload
+    end_bytes, temp_bytes = splitBytesOnDelimiter(msg.payload)
+    player_bytes, win_bytes = splitBytesOnDelimiter(temp_bytes)
+    # convert the bytes back into useable variables / structures
+    end_status = end_bytes.decode()
+    win_player = player_bytes.decode()
+    win_method = win_bytes.decode()
+    return end_status, win_player, win_method
+
+
+
+
+
+
+# ========================= TEMP TEST CODE PLZ IGNORE ==============================
```

Comment:
noting existence of test code

Reply:
The only way the donny knows how to program is trial and error lol